src/lib.rs

```diff
@@ -802,102 +795,25 @@ pub enum BindingResource<'a> {
-    pub bind_group_layouts: &'a [&'a BindGroupLayout],
-}
+pub type PipelineLayoutDescriptor<'a> = wgt::PipelineLayoutDescriptor<'a, &'a BindGroupLayout>;
 
 /// Describes a programmable pipeline stage.
-#[derive(Clone)]
-pub struct ProgrammableStageDescriptor<'a> {
-    /// The compiled shader module for this stage.
-    pub module: &'a ShaderModule,
-
-    /// The name of the entry point in the compiled shader. There must be a function that returns
-    /// void with this name in the shader.
-    pub entry_point: &'a str,
-}
+pub type ProgrammableStageDescriptor<'a> = wgt::ProgrammableStageDescriptor<'a, &'a ShaderModule>;
 
 /// Describes a render (graphics) pipeline.
-#[derive(Clone)]
-pub struct RenderPipelineDescriptor<'a> {
-    /// The layout of bind groups for this pipeline.
-    pub layout: &'a PipelineLayout,
-
-    /// The compiled vertex stage and its entry point.
-    pub vertex_stage: ProgrammableStageDescriptor<'a>,
-
-    /// The compiled fragment stage and its entry point, if any.
-    pub fragment_stage: Option<ProgrammableStageDescriptor<'a>>,
-
-    /// The rasterization process for this pipeline.
-    pub rasterization_state: Option<RasterizationStateDescriptor>,
-
-    /// The primitive topology used to interpret vertices.
-    pub primitive_topology: PrimitiveTopology,
-
-    /// The effect of draw calls on the color aspect of the output target.
-    pub color_states: &'a [ColorStateDescriptor],
-
-    /// The effect of draw calls on the depth and stencil aspects of the output target, if any.
-    pub depth_stencil_state: Option<DepthStencilStateDescriptor>,
-
-    /// The vertex input state for this pipeline.
-    pub vertex_state: VertexStateDescriptor<'a>,
-
-    /// The number of samples calculated per pixel (for MSAA). For non-multisampled textures,
-    /// this should be `1`
-    pub sample_count: u32,
-
-    /// Bitmask that restricts the samples of a pixel modified by this pipeline. All samples
-    /// can be enabled using the value `!0`
-    pub sample_mask: u32,
-
-    /// When enabled, produces another sample mask per pixel based on the alpha output value, that
-    /// is ANDed with the sample_mask and the primitive coverage to restrict the set of samples
-    /// affected by a primitive.
-    ///
-    /// The implicit mask produced for alpha of zero is guaranteed to be zero, and for alpha of one
-    /// is guaranteed to be all 1-s.
-    pub alpha_to_coverage_enabled: bool,
-}
+pub type RenderPipelineDescriptor<'a> =
+    wgt::RenderPipelineDescriptor<'a, &'a PipelineLayout, ProgrammableStageDescriptor<'a>>;
 
 /// Describes a compute pipeline.
-#[derive(Clone)]
-pub struct ComputePipelineDescriptor<'a> {
-    /// The layout of bind groups for this pipeline.
-    pub layout: &'a PipelineLayout,
-
-    /// The compiled compute stage and its entry point.
-    pub compute_stage: ProgrammableStageDescriptor<'a>,
-}
+pub type ComputePipelineDescriptor<'a> =
```

Comment:
For the other descriptor types, we included the the wgt type as XDescriptorBase for docs purposes. Do we still want to do that? If so, these should be imported that way too.

Reply:
Oh right, I totally missed that. Fully agree that we need to be consistent with regards to naming here.
Either stop using the `Base` suffix, or use it everywhere. Would the latter cause any problems to the docs? If not, I'm fine with either.
This would need to be changed in wgpu though, so technically this PR to wgpu-rs doesn't need to be blocked. We can get it in, and then just update the naming here.